### PR TITLE
chore(baseline): update learn more link to glossary entry

### PR DESCRIPTION
**Pull request by LeoMcA**
## Summary

https://mozilla-hub.atlassian.net/browse/MP-1272

### Problem

Baseline banner was linking to rather old blog post

### Solution

Link to the maintained, and up to date glossary entry

Disabled noreferrer rule for link, as it's internal (eslint can't detect this because its dynamic), and it's actually useful to have referrer data here.

---

## Screenshots

No change

---

## How did you test this change?

- `yarn && yarn dev`
- localhost:3000
- tested across locales: those with a glossary page go there directly, those without recommend going to the english glossary page

### CODE REVIEW

File: client/src/document/baseline-indicator.tsx [hunk 1]
```diff
@@ -174,8 +174,9 @@ export function BaselineIndicator({ status }: { status: SupportStatus }) {
         )}
         <ul>
           <li>
+            {/* eslint-disable-next-line react/jsx-no-target-blank */}
```
Comment (caugner) [inline client/src/document/baseline-indicator.tsx:177]: Why is this needed?

Comment (LeoMcA) [reply client/src/document/baseline-indicator.tsx:177]: eslint can't analyze dynamic links to determine if they're local or external, so fails safe and enables the rule for them: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-target-blank.md#enforcedynamiclinks

I reckon it would be useful to have referrer data on the glossary page to see which pages users are coming from

Comment (caugner) [reply client/src/document/baseline-indicator.tsx:177]: Okay, all good.

**Review by caugner — APPROVED** [review]
LGTM, although for locales where that Baseline Glossary page isn't written, it will link to a 404 page, but there's no easy fix.
